tests/foreman/api/test_classparameters.py

```diff
@@ -0,0 +1,1035 @@
+        6.  Set 'merge overrides' to True.
+        7.  Set 'merge default' to True.
+        8.  Update the parameter with above steps.
+        9.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all
+        the associated matchers.
+        2.  The YAML output doesn't have the empty default value of parameter.
+        3.  Duplicate values in YAML output if any are displayed.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_merge_puppet_default(self):
+        """Merge the values of all the associated matchers + puppet default value.
+
+        @feature: Class Parameters - Matcher
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Set default Value as puppet default value.
+        3.  Create first matcher for attribute fqdn with valid details.
+        4.  Create second matcher for other attribute with valid details.
+        Note - The fqdn/host should have this attribute.
+        5.  Create more matchers for some more attributes if any.
+        Note - The fqdn/host should have this attributes.
+        6.  Set 'merge overrides' to True.
+        7.  Set 'merge default' to True.
+        8.  Update the parameter with above steps.
+        9.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all
+        the associated matchers.
+        2.  The YAML output doesn't have the puppet default value.
+        3.  Duplicate values in YAML output if any are displayed.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_avoid_duplicate(self):
+        """Merge the values of all the associated matchers, remove duplicates.
+
+        @feature: Class Parameters - Matcher
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Set some default Value of array type.
+        3.  Create first matcher for attribute fqdn with some value.
+        4.  Create second matcher for other attribute with
+        same value as fqdn matcher.
+        Note - The fqdn/host should have this attribute.
+        5.  Set 'merge overrides' to True.
+        6.  Set 'merge default' to True.
+        7.  Set 'avoid duplicate' to True.
+        8.  Update the parameter with above steps.
+        9.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all
+        the associated matchers.
+        2.  The YAML output has the default value of parameter.
+        3.  Duplicate values in YAML output are removed / not displayed.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create_matcher_avoid_duplicate(self):
+        """Duplicates not removed as they were not really present.
```

Comment:
a little bit strange test case, but probably still valid. Anyway, please use
`Duplicates are not removed as they were not really present.`